pkg/apis/core/v1alpha1/defaults.go

```diff
@@ -231,6 +231,10 @@ func SetDefaults_Shoot(obj *Shoot) {
 	if obj.Spec.Maintenance == nil {
 		obj.Spec.Maintenance = &Maintenance{}
 	}
+
+	if obj.Spec.Kubernetes.KubeAPIServer.EnableAnonymousAuthentication == nil {
+		obj.Spec.Kubernetes.KubeAPIServer.EnableAnonymousAuthentication = pointer.BoolPtr(false)
+	}
```

Comment:
I vote to remove the explicit defaulting. You already have `ShootWantsAnonymousAuthentication` that handles the case when the field is not defined.

Reply:
In the past, we discussed this topic (defaulting optional fields explicitly) and IIRC then the tendency was more in the direction of indeed explicitly default newly added fields because it makes it more clear how certain values are set.

Reply:
I left it as it is and provided tests that check that defaulting does not disable explicitly set values.